cpp/pipelines/add_authority_external_ref.cc

```diff
@@ -18,120 +19,201 @@
+                        }
+                    }
                 }
-                act_gnd = "";
-                act_name = "";
-                act_wikidata = "";
-                act_wikipedia = "";
-                sameAs_reached = false;
-                read_gnd_id = true;
-            } else if (read_preferred_name) {
-                read_preferred_name = false;
-                act_name = std::regex_replace(line, std::regex("(value|:|\"|@)"), "");
-                act_name = StringUtil::TrimWhite(act_name);
-            } else if (StringUtil::Contains(line, "info/gnd/") and read_gnd_id) {
-                read_gnd_id = false;
-                std::size_t last_slash = line.find_last_of("/");
-                act_gnd = line.substr(last_slash + 1);
-                act_gnd = std::regex_replace(act_gnd, std::regex("(\\s|,|\")"), "");
-            } else if (StringUtil::Contains(line, "www.wikidata.org/entity/") and sameAs_reached) {
-                std::size_t last_slash = line.find_last_of("/");
-                act_wikidata = line.substr(last_slash + 1);
-                act_wikidata = std::regex_replace(act_wikidata, std::regex("(\\s|,|\")"), "");
-            } else if (StringUtil::Contains(line, "wikipedia.org/wiki/") and StringUtil::Contains(line, "http") and sameAs_reached) {
-                std::size_t first_http = line.find("http");
-                act_wikipedia = line.substr(first_http);
-                act_wikipedia = std::regex_replace(act_wikipedia, std::regex("(\\s|,|\")"), "");
-            } else if (StringUtil::Contains(line, "owl#sameAs")) {
-                sameAs_reached = true;
-            } else if (StringUtil::Contains(line, "preferredNameForThePerson")) {
-                read_preferred_name = true;
             }
         }
-        input_file.close();
-        output_file.close();
-    } else
-        LOG_ERROR("input or output files could not be opened");
-}
+        std::cout << "\r"
+                  << "Parsed: " << total_line_parsed << " line(s), "
+                  << " Total GND-ID: " << total_numbers_of_gnd_id_generated << ", Total GND with Wikidata: " << total_number_of_wikidata
+                  << ", Total GND with Wikipedia: " << total_number_of_wikipedia;
+        //   << "GND-ID url: " << tmp_id;
+        std::cout.flush();
+        ++total_line_parsed;
+    }
+
+    const auto load_file_end(std::chrono::high_resolution_clock::now());
+    auto duration = std::chrono::duration_cast<std::chrono::seconds>(load_file_end - load_file_start);
 
+    std::cout << std::endl
+              << "Total GND-ID: " << total_numbers_of_gnd_id_generated << std::endl
+              << "Total GND with Wikidata: " << total_number_of_wikidata << std::endl
+              << "Total GND with Wikipedia: " << total_number_of_wikipedia << std::endl;
+    std::cout << "Total time of computation: " << duration.count() << " second(s)" << std::endl;
 
+    input_file.close();
+    return true;
+}
```

Comment:
By the way, do you have the git hooks enabled? It seems like the amount of empty newlines before GarseGndWikidataMappingFile does not match our convention.